src/persistence.rs

```diff
@@ -69,18 +68,11 @@ mod serde_ark_unchecked {
 struct KeystoreSnapshot<L: Ledger> {
     txn_state: TransactionState<L>,
     key_state: KeyStreamState,
```

Comment:
Rather than leaving this here, I feel like it belongs with `Accounts` somehow, but it's a little weird since it's state for the whole `Accounts` resource and not any particular `Account`.

One suggestion: add a field `Account::index: Option<usize>` which is the index into the HD key stream of this key, for generated keys, and `None` for imported keys (note: MetaMask does similar bookkeeping, it will tell you the index of your accounts and which accounts are imported). Then we don't need to store `KeyStreamState` at all. The index of the next HD key is simply `accounts.iter().filter_map(|account| account.index).max() + 1`.

Reply:
Can we add `index` to `Accounts` instead of `Account` to avoid the `filter_map` when looking up the maximum index? Since we only need the maximum value, I think we don't have to store other indices.

Reply:
Oh, like make it a field on `Accounts` and populate it when you load from storage by doing the `filter_map` just once? That sounds like a good idea

Reply:
Cool, I'll make the change then!